### PR TITLE
fix favorites acceptance tests

### PR DESCRIPTION
**Pull request by individual-it**
## Description
1. fix snake oil tests that were not doing what they promised (pagination does not work and was never implemented)
2. skip tests that we know will fail
3. move favourites related steps to own context

## Related Issue
part of #33690

## Motivation and Context
make WebDav.php shorter

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/apiFavorites/favorites.feature [hunk 1]
```diff
@@ -94,7 +96,28 @@ Feature: favorite
...
+  @skip @issue-33840
+  Scenario Outline: Get favorited elements and limit count of entries
+    Given using <dav_version> DAV path
+    And the user has favorited element "/textfile0.txt"
+    And the user has favorited element "/textfile1.txt"
+    And the user has favorited element "/textfile2.txt"
+    And the user has favorited element "/textfile3.txt"
+    And the user has favorited element "/textfile4.txt"
+    When the user lists the favorites of folder "/" and limits the result to 3 elements using the WebDAV API
+    And the search result of "user0" shoud contain any "3" of these entries:
```
Comment (phil-davis) [inline tests/acceptance/features/apiFavorites/favorites.feature]: s/And/Then

File: tests/acceptance/features/apiFavorites/favorites.feature [hunk 2]
```diff
@@ -103,15 +126,19 @@ Feature: favorite
...
-    And the user favorites element "/subfolder/textfile5.txt" using the WebDAV API
-    Then the user in folder "/subfolder" should have favorited the following elements from offset 3 and limit 2
+    And the user has favorited element "/subfolder/textfile0.txt"
+    And the user has favorited element "/subfolder/textfile1.txt"
+    And the user has favorited element "/subfolder/textfile2.txt"
+    And the user has favorited element "/subfolder/textfile3.txt"
+    And the user has favorited element "/subfolder/textfile4.txt"
+    And the user has favorited element "/subfolder/textfile5.txt"
+    When the user lists the favorites of folder "/" and limits the result to 3 elements using the WebDAV API
+    And the search result of "user0" shoud contain any "3" of these entries:
```
Comment (phil-davis) [inline tests/acceptance/features/apiFavorites/favorites.feature]: s/And/Then

**Review by phil-davis — DISMISSED** [review]
looks OK, let's see what CI says

**Comment by individual-it** [comment]
backport in #33852

**Review by phil-davis — APPROVED** [review]

**Pull request merged**